packages/components/src/Modal/_docs/Modal.stories.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { useState } from "react"
+import { StoryFn } from "@storybook/react"
+import { Button } from "@kaizen/button"
+import { Paragraph } from "@kaizen/typography"
+import { Modal } from "../"
+import { modalControls } from "./controls/modalControls"
+
+export default {
+  title: "Components/Modal",
+  component: Modal,
+  parameters: {
+    actions: {
+      argTypesRegex: "^on.*",
+    },
+  },
+  argTypes: {
+    ...modalControls,
+  },
+}
+
+export const DefaultStory: StoryFn<typeof Modal> = props => {
+  const [modalOpen, setModalOpen] = useState<boolean>(false)
+  return (
+    <>
+      <Button label="Open modal" onClick={(): void => setModalOpen(true)} />
+      <Modal
+        {...props}
+        isOpen={modalOpen}
+        onDismiss={(): void => setModalOpen(false)}
+      >
+        <Modal.Header>
+          <Modal.Title id="my-modal-title">Generic content modal</Modal.Title>
+        </Modal.Header>
```

Comment:
Can we consider a more Prop driven component on top of this?

Something like:
```
<Modal title="" actions=[{ ...buttonProps}, { ...buttonProps }]>
  ...
</Modal>
```

This can act as the top level component for straightforward modal cases and we can still expose all of these compositional components so they can create what you have here when they need it.